### ISSUE TITLE
Firefox 57 / Greasemonkey 4 userscript

### ISSUE THREAD

**Issue by Pantalaim0n** [Aktanusa/CookieMonster#153]
The Greasemonkey userscript stopped working for me today after upgrading to firefox 57 (Quantum) / Greasemonkey 4

I made some changes to get it to work again:
```
// ==UserScript==
// @name Cookie Monster
// @namespace Cookie
// @include http://orteil.dashnet.org/cookieclicker/
// @version 1
// @grant none
// ==/UserScript==

javascript:(function() {
    var checkReady = setInterval(function() {
        if (typeof unsafeWindow.Game.ready !== 'undefined' && unsafeWindow.Game.ready) {
            unsafeWindow.Game.LoadMod('http://aktanusa.github.io/CookieMonster/CookieMonster.js');
            clearInterval(checkReady);
        }
    }, 1000);
}());
``` 

The `Game` object is undefined, but with the prefix `unsafeWindow` it works again.

**Comment by Aktanusa** [comment]
Hmm, interesting.  Do you know if it works fine with other browsers with your changes like Google Chrome?

**Comment by Pantalaim0n** [comment]
It works in Chrome / Tampermonkey if I remove the line `// @grant none`.
Removing the same line in FireFox makes no difference, it works in both cases.

**Comment by Aktanusa** [comment]
Thanks for the research.  I'll take a look if there is another way instead of using `unsafeWindow` since apparently it is not recommended to use it.

**Comment by cpeter256** [comment]
I've been using the following script successfully-- window.eval appears to be the recommended way to run code inside the security context of the page, instead of greasemonkey's context

```javascript
var code = "(" + (function() {
  var checkReady = setInterval(function() {
    if (typeof Game.ready !== 'undefined' && Game.ready) {
      Game.LoadMod('http://aktanusa.github.io/CookieMonster/CookieMonster.js');
      clearInterval(checkReady);
    }
  }, 1000);
}).toString() + ")()";

window.eval(code);
```

**Comment by Aktanusa** [comment]
thanks!  I'll will probably use that instead of `unsafeWindow`.  Should be updated when I updated CM to latest CC.

**Comment by Aktanusa** [comment]
Fixed in version 2.0045.1
